### ISSUE TITLE
Rishi's AP FRQs Review Ticket

### ISSUE THREAD

**Issue by rpeddakama** [rpeddakama/Tri-2-CSA-Project#51]
# Rishi's AP FRQs
## Important Stuff
| Topic | Description |
| - | - |
| [9 FRQs](https://dnhsnest.tk/rishi) | All Frq's can be found on my about page |
| [Takeaways](https://github.com/rpeddakama/Tri-2-CSA-Project/wiki/Rishi-Peddakama---FRQ-Takeaways) | A comprehensive wiki with learning points and tangibles for each frq |
| FRQ Source Code [Backend](https://github.com/rpeddakama/Tri-2-CSA-Project/blob/master/src/main/java/com/example/sping_portfolio/controllers/Rishi.java), [Frontend](https://github.com/rpeddakama/Tri-2-CSA-Project/blob/master/src/main/resources/templates/rishi.html) | The Backend for All Frqs are in the same Rishi.java file and are all displayed in the rishi.html file |
| [Partner Grading](https://github.com/ChristopherDuroiu/usb-c/issues/31#issuecomment-1031735892) | The comment I made on Chris's review ticket. Contains a grade out of 10 |

Most of these Frqs have input feature that can be tried out on the [runtime](https://dnhsnest.tk/rishi).

## Crossover Comparison with Christopher Duroiu
| Frq in Comparison | Chris's Frq | Rishi's Frq | Evaluation/Comparison |
| - | - | - | - |
| 4 | [Runtime](http://dnstonks.cf/frq4), [Code](https://github.com/ChristopherDuroiu/usb-c/blob/main/src/main/java/com/example/sping_portfolio/controllers/chris/LongestChar.java) | [Runtime](https://dnhsnest.tk/rishi), [Code](https://github.com/rpeddakama/Tri-2-CSA-Project/blob/master/src/main/java/com/example/sping_portfolio/controllers/Rishi.java#L466) | Outside of final string formatting, both of our runtimes produced the same answer. Also, in situations where a string have more than one substring of max length, both of our algorithms returned the first substring. The biggest difference in our actual code was that Chis kept track of the value of the longest string while I only kept track of the character. One thing Chris could do to improve runtime is to only keep track of the character is it reduces the amount of memory that needs to be allocated. |
| 6 | [Runtime and Code](https://replit.com/@ChrisD9/ingidentifier#Main.java) | [Runtime](https://dnhsnest.tk/rishi), [Code](https://github.com/rpeddakama/Tri-2-CSA-Project/blob/master/src/main/java/com/example/sping_portfolio/controllers/Rishi.java#L500) | For this Frq, mine was implemented in the PBL project with input while Chris did his in repl without input. I think my code is concise and efficient because there is only 1 for loop and it utilizes the .substring() function to get the last three characters of the string. Chris's on the other hand, iterated over the string backwards to get the last three characters and check its values in reverse: "gni". Though it still gets the same result, some of the logic is hard to follow. Nonetheless, I never thought of approaching the problem in this way and I think viewing his code helped with my problem solving. |
| 9 | [Runtime and Code](https://replit.com/@ChrisD9/SkinnyMulticoloredSpyware#Main.java) | [Runtime](https://dnhsnest.tk/rishi), [Code](https://github.com/rpeddakama/Tri-2-CSA-Project/blob/master/src/main/java/com/example/sping_portfolio/controllers/Rishi.java#L530) | This one of the more simpler FRQ's, focused on demonstrating Java's child, parent, and abstract classes. Thus, it is no surprise that our code looks very similar other than our different coding style. Mine was in PBL his was in repl (both no input) |

## Summary
Overall, I think these FRQs helped me solidify the content on the AP Computer Science A exam. With a little review here and there, I should be prepared to do well on the AP Exam. From the comparisons with my crossover partner, I saw the different ways people approach a problem and how they code to solve it accordingly.

**Comment by ChristopherDuroiu** [comment]
# Crossover Grading

**Total score: 9.9/10**
Unit 2: 1/1
Unit 3: 1/1
Unit 4: 1/1
Unit 5: 1/1
Unit 6: 1/1
Unit 7: 1/1
Unit 8: 1/1
Unit 9: 1/1
Unit 10: 1/1
At least 2 inputs: 1/1

Comments: All of Rishi's FRQs are made highly efficient and majority have inputs, showing he went beyond what was asked. He also has comments on some parts of the code to make reading it easier.

Other grading:
Rishi's wiki includes everything necessary, though he could include more descriptive takeaways for the FRQs, instead of just a list of the topics.
